simulator.py

```diff
@@ -0,0 +1,98 @@
+"""
+Simulator that simulates trading process and estimates profits and losses.
+"""
+
+import copy
+import uuid
+from thousandaire.dataset import Data
+
+def decode_data(data):
+    """
+    Decode while in multi-process.
+
+    In multi-processing, the simulator will get data by shared memory, which
+    are encoded by pickle to make them sharable.
+    To avoid polluting the shared memory, we need to create a local copy.
+    Since 'price_data' may also be in 'others', we should deepcopy the data
+    separately instead of using copy.deepcopy(data).
+    """
+    return {
+        'workdays' : copy.deepcopy(data['workdays']),
+        'price_data' : copy.deepcopy(data['price_data']),
+        'others' : copy.deepcopy(data['others'])}

```

Comment:
While these are fine in the current version, since we still have to manually sync some magic constants (like three constant strings here), in the long run this may not be a good practice.
It is possible that we have to pass more arguments from *simulation* to *simulator* in the future.
At the time, we can create a new class to define all these "message", which can make synchronization easier.